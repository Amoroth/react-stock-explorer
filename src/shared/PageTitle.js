import React from 'react'
import { withRouter } from 'react-router-dom'

import SmallSpinner from './SmallSpinner'
import styles from './shared.module.css'

const pageTitle = (props) => {
  const onTitleClick = () => {
    if (props.link) {
      props.history.push(`/company?cmp=${props.symbol}`)
    }
  }

  const onBackClick = () => {
    if (props.link) {
      props.history.push('/')
    } else {
      props.history.goBack()
    }
  }

  return (
    <div className={styles['title-bar']}>
      <div style={{ display: 'flex' }}>
        {props.logo ? 
        <img
          src={props.logo}
          height={56}
          alt={`${props.name}'s logo`}
          style={{ maxWidth: '100px' }}
        /> :
        <SmallSpinner />}
        <div>
          <h6
            onClick={ onTitleClick }
            style={{ cursor: props.link ? 'pointer' : 'default' }}
          >{ props.name }</h6>
          <span>{ props.exchange }: { props.symbol }</span>
        </div>
      </div>
      <button onClick={ onBackClick }>
        <i className="material-icons md-48">arrow_back</i>
      </button>
    </div>
  )
}

export default withRouter(pageTitle)
