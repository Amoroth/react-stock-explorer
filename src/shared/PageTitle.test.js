import React from 'react'
import { shallow } from 'enzyme'

import PageTitle from './PageTitle'

describe('<PageTitle />', () => {
  let props
  let wrapper

  beforeEach(() => {
    props = {
      logo: 'https://placekitten.com/g/56/56',
      name: 'Company',
      link: false,
      symbol: 'COM',
      exchange: 'PRI',
      history: { push: jest.fn(), goBack: jest.fn() },
      favorite: false,
      onFav: jest.fn(),
    }
    wrapper = shallow(<PageTitle.WrappedComponent {...props} />)
  })

  it('renders', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders title without link prop', () => {
    expect(wrapper.find('h6').exists()).toBe(true)
    expect(wrapper.find('Link').exists()).toBe(false)
    expect(wrapper.find('h6').text()).toBe(props.name)
  })

  it('renders link with link prop', () => {
    wrapper.setProps({ link: true })

    expect(wrapper.find('h6').exists()).toBe(false)
    expect(wrapper.find('Link').exists()).toBe(true)
    expect(wrapper.find('Link').children().first().text()).toBe(props.name)
  })

  it('renders symbol and exchange', () => {
    expect(wrapper.find('span').text()).toBe(`${props.exchange}: ${props.symbol}`)
  })

  it('sets up logo', () => {
    expect(wrapper.find('img').prop('src')).toBe(props.logo)
    expect(wrapper.find('img').prop('alt')).toBe(`${props.name}'s logo`)
  })

  it('shows spinner if no logo', () => {
    wrapper.setProps({ logo: null })

    expect(wrapper.find('smallSpinner').exists()).toBe(true)
    expect(wrapper.find('img').exists()).toBe(false)
  })

  it('sets favorite', () => {
    wrapper.find('.favorite-button').simulate('click')
    expect(props.onFav).toBeCalled()
    expect(props.onFav.mock.calls[0][1]).toBe(props.symbol)
  })

  it('changes logo if favorite', () => {
    let button = wrapper.find('.favorite-button').children().first()
    expect(button.text()).toBe('favorite_border')
    wrapper.setProps({ favorite: true })
    button = wrapper.find('.favorite-button').children().first()
    expect(button.text()).toBe('favorite')
    expect(button.prop('style')).toEqual({ color: 'red' })
  })

  it('back button sends to market if link prop', () => {
    wrapper.setProps({ link: true })
    wrapper.find('.back-button').simulate('click')
    expect(props.history.push).toBeCalledWith('/market')
  })

  it('back button sends to previous if no link prop', () => {
    wrapper.find('.back-button').simulate('click')
    expect(props.history.goBack).toBeCalled()
  })
})
