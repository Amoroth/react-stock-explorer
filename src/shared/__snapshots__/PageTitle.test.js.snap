// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageTitle /> renders 1`] = `
<div
  className="title-bar"
>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <img
      alt="Company's logo"
      height={56}
      src="https://placekitten.com/g/56/56"
      style={
        Object {
          "maxWidth": "100px",
        }
      }
    />
    <div>
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <h6>
          Company
        </h6>
        <button
          className="favorite-button"
          onClick={[Function]}
          type="button"
        >
          <i
            className="material-icons"
          >
            favorite_border
          </i>
        </button>
      </div>
      <span>
        PRI: COM
      </span>
    </div>
  </div>
  <button
    className="back-button"
    onClick={[Function]}
    type="button"
  >
    <i
      className="material-icons md-48"
    >
      arrow_back
    </i>
  </button>
</div>
`;
